### PR TITLE
 bugfix 995

### PR DESCRIPTION
**Pull request by gluk256**

### CODE REVIEW

File: libp2p/NodeTable.cpp [hunk 1]
```diff
@@ -199,7 +199,16 @@ void NodeTable::doDiscover(NodeId _node, unsigned _round, shared_ptr<set<shared_
 	m_timers.schedule(c_reqTimeout.count() * 2, [this, _node, _round, _tried](boost::system::error_code const& _ec)
 	{
 		if (_ec)
-			clog(NodeTableWarn) << "Discovery timer canceled!";
+			clog(NodeTableWarn) << "Discovery timer canceled: " << _ec.value() << _ec.message();
+
+		if (995 == _ec.value() || m_timers.isStopped())
```
Comment (chriseth) [inline libp2p/NodeTable.cpp]: Is there a name for `995`? If yes, please use that.

Comment (gavofyork) [inline libp2p/NodeTable.cpp]: indeed. right now this just looks like a backdoor!

also, while i'm aware of the reason for the awkward <constant> == <variable> ordering, the rest of the codebase uses <variable> == <constant> ordering. consistency is king.

**Comment by gluk256** [comment]
fixed

File: libp2p/NodeTable.cpp [hunk 2]
```diff
@@ -199,7 +199,17 @@ void NodeTable::doDiscover(NodeId _node, unsigned _round, shared_ptr<set<shared_
 	m_timers.schedule(c_reqTimeout.count() * 2, [this, _node, _round, _tried](boost::system::error_code const& _ec)
 	{
 		if (_ec)
-			clog(NodeTableWarn) << "Discovery timer canceled!";
+			clog(NodeTableWarn) << "Discovery timer canceled: " << _ec.value() << _ec.message();
```
Comment (gavofyork) [inline libp2p/NodeTable.cpp]: 0 level of required verbosity (`NodeTableWarn::verbosity`) for this message is too low.

all developer-level notices should be 2 or greater. for low-level stuff like this 4 or greater.

**Comment by gavofyork** [comment]
verbosity reduction needed.

**Pull request merged**